src/app/features/psbt-signer/hooks/use-psbt-decoded-request.tsx

```diff
@@ -4,29 +4,41 @@ import * as btc from '@scure/btc-signer';
 
 import { BitcoinNetworkModes } from '@shared/constants';
 import { getAddressFromOutScript } from '@shared/crypto/bitcoin/bitcoin.utils';
-import { isUndefined } from '@shared/utils';
 
-import {
-  OrdApiInscriptionTxOutput,
-  useOrdinalsAwareUtxoQueries,
-} from '@app/query/bitcoin/ordinals/ordinals-aware-utxo.query';
 import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
 import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
 
+import {
+  PsbtDecodedUtxosMainnet,
+  PsbtDecodedUtxosTestnet,
+  usePsbtDecodedUtxos,
+} from './use-psbt-decoded-utxos';
+
 function isPlaceholderTransaction(
   address: string,
-  inputs: (OrdApiInscriptionTxOutput | undefined)[],
-  outputs: btc.TransactionOutputRequired[],
-  network: BitcoinNetworkModes
+  network: BitcoinNetworkModes,
+  unsignedOutputs: btc.TransactionOutputRequired[],
+  unsignedUtxos: PsbtDecodedUtxosMainnet | PsbtDecodedUtxosTestnet
 ) {
-  const inputsNotFromCurrentAddress = inputs.filter(input => {
-    return input?.address !== address;
-  });
-  const outputsNotToCurrentAddress = outputs.filter(output => {
+  let utxosNotFromCurrentAddress = [];
+
+  switch (unsignedUtxos.network) {
+    case 'mainnet':
+      utxosNotFromCurrentAddress = unsignedUtxos.utxos.filter(utxo => utxo.address !== address);
+      break;
+    case 'testnet':
+      utxosNotFromCurrentAddress = unsignedUtxos.utxos.filter(
+        vo => vo.scriptpubkey_address !== address
+      );
+      break;
+  }
```

Comment:
Could write without reassigning var

```suggestion
  const utxosNotFromCurrentAddress = unsignedUtxos.filter(utxo => whenNetwork(unsignedUtxos.network)({
    mainnet: utxo.address !== address,
    testnet: vo.scriptpubkey_address !== address
  }));
```

Reply:
Does this work with the types being different so no errors? Will check bc thought I had to use the switch with the discriminating union here to avoid type errors.

Reply:
Don't think works...

<img width="1808" alt="Screen Shot 2023-06-16 at 11 21 05 AM" src="https://github.com/hirosystems/wallet/assets/6493321/65f2275c-0265-40e7-ba61-6f812f7e493c">

Reply:
Sure, see that the the `when` helper doesn't narrow the types. 

I wonder though, earlier in the data flow, could we not just make sure both utxo types have a property called `address` circumventing the need for this conditional entirely?